### DISCUSSION TITLE
Custom theme colors.

### DISCUSSION THREAD

**Discussion by ealmiladi** [nativewind/nativewind#392] [Q&A] [answered]
Our application allows users to select between a set of themes, e.g. dark, light, warm, cool, etc. On web using TailwindCSS, we're able to get away with this by utilizing what TailwindCSS suggests:

```
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  :root {
    --color-primary: 255 115 179;
    --color-secondary: 111 114 185;
    /* ... */
  }

  .warm {
    --color-primary: 111 225 339;
    --color-secondary: 252 252 252;
    /* ... */
  }
}
```

and then in `tailwind.config.js`:

```
module.exports = {
  theme: {
    colors: {
      // Using modern `rgb`
      primary: 'rgb(var(--color-primary) / <alpha-value>)',
      secondary: 'rgb(var(--color-secondary) / <alpha-value>)',

      // Using modern `hsl`
      primary: 'hsl(var(--color-primary) / <alpha-value>)',
      secondary: 'hsl(var(--color-secondary) / <alpha-value>)',

      // Using legacy `rgba`
      primary: 'rgba(var(--color-primary), <alpha-value>)',
      secondary: 'rgba(var(--color-secondary), <alpha-value>)',
    }
  }
}
```

How might we achieve the same thing using NativeWind? We don't want to have to specify a variant for each theme, i.e. `<View className="dark:bg-dark-500 warm:bg-red-500"/>` etc.

**Comment by ealmiladi** [comment, marked as answer]
Found https://github.com/marklawlor/nativewind/issues/308 which I believe addresses this.

**Comment by danstepanov** [comment]
Closing this discussion, if this is still relevant, feel free to open a new one.
